### PR TITLE
[MCH-283-141] Feat: solidly dex uses hop tokens

### PR DESCRIPTION
**Pull request by gotzenx**

### CODE REVIEW

File: packages/yswaps/scripts/libraries/dexes/solidly.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ export type SwapResponse = {
 };
 
 export const getBestPathEncoded = async (swapParams: SwapParams): Promise<SwapResponse> => {
+  const { tokenIn, tokenOut, hopTokensToTest, amountIn } = swapParams;
   swapParams.hopTokensToTest = swapParams.hopTokensToTest ?? [];
```
Comment (b0dhidharma) [inline packages/yswaps/scripts/libraries/dexes/solidly.ts:25]:
```suggestion
  const { tokenIn, tokenOut, amountIn } = swapParams;
```
Not going to be used.

File: packages/yswaps/scripts/libraries/dexes/solidly.ts [hunk 2]
```diff
@@ -75,6 +65,92 @@ export const getBestPathEncoded = async (swapParams: SwapParams): Promise<SwapRe
   return response;
 };
 
+
+async function hopTokenCalculation({ swapParams }: {swapParams: SwapParams}) {
+  const { tokenIn, tokenOut, amountIn, hopTokensToTest } = swapParams;
```
Comment (b0dhidharma) [inline packages/yswaps/scripts/libraries/dexes/solidly.ts:70]:
```suggestion
  const { tokenIn, tokenOut, hopTokensToTest, amountIn } = swapParams;
 hopTokensToTest = hopTokensToTest ?? [];
```

File: packages/yswaps/scripts/libraries/dexes/solidly.ts [hunk 3]
```diff
@@ -75,6 +65,92 @@ export const getBestPathEncoded = async (swapParams: SwapParams): Promise<SwapRe
...
+  const router = await ethers.getContractAt<ISolidlyRouter>(ISolidlyRouter__factory.abi, swapParams.solidlyRouter);
+
+  let bestMaxPath: ISolidlyRouter.RouteStruct[] = []
+  let bestAmountOut: BigNumber = BigNumber.from('0');
+  let bestHopToken: string = '';
+  if (!hopTokensToTest) return { bestMaxPath, bestAmountOut };
+
+  for (const hopToken of hopTokensToTest) {
+    let hopMaxPath: ISolidlyRouter.RouteStruct[] = [];
+    let hopAmountOut: BigNumber = BigNumber.from('0');
```
Comment (b0dhidharma) [inline packages/yswaps/scripts/libraries/dexes/solidly.ts:81]: There is no need of creating this two I think. You can use `bestMaxPath` and `bestAmountOut`

Comment (gotzenx) [reply packages/yswaps/scripts/libraries/dexes/solidly.ts]: correct, also I just refactored to not duplicate the IF logic to assign amountOut and maxPath

**Review by b0dhidharma — APPROVED** [review]

**Pull request merged**